contracts/src/7683/T1ERC7683.sol

```diff
@@ -419,15 +424,80 @@ contract T1ERC7683 is IT1ERC7683, T1Permit2, AccessControlUpgradeable, EIP712 {
     function handleReadResultWithProof(bytes calldata encodedProofOfRead) external whenSettleNotPaused {
         (bytes32 requestId, bytes memory result) = xChainRead.verifyProofOfRead(encodedProofOfRead);
 
-        bytes32 orderId = settlementReadRequestToOrderId[requestId];
+        (bytes32 orderId, bool isSettled, address inputToken, address settlementReceiver, uint256 amount) =
+            _decodeOrdersToSettle(requestId, result);
+
+        if (isSettled) {
+            _transferTokenOut(inputToken, settlementReceiver, amount);
+        }
+
+        emit SettlementVerified(orderId, isSettled);
+    }
+
+    /// @notice Use result of proof of read to handle batch of orders depending on the result
+    /// Also enforce auction winner bid if the orderId has closed auction.
+    /// @param encodedProofsOfRead The encoded proofs of read which are formatted as following:
+    /// abi.encode(uint256 batchIndex, bytes32 requestId, uint256 position, bytes result, bytes proof)
+    function handleBatchOfReadResultsWithProofs(bytes[] calldata encodedProofsOfRead) external whenSettleNotPaused {
+        (bytes32[] memory requestIds, bytes[] memory results) = xChainRead.verifyProofsOfRead(encodedProofsOfRead);
+
+        bytes32[] memory orderIds = new bytes32[](encodedProofsOfRead.length);
+        bool[] memory areSettled = new bool[](encodedProofsOfRead.length);
+        UserTokens[] memory userTokenKeys = new UserTokens[](encodedProofsOfRead.length);
+        uint256[] memory amounts = new uint256[](encodedProofsOfRead.length);
+        uint256 uniqueUserTokenCount = 0;
+
+        for (uint256 i = 0; i < encodedProofsOfRead.length; i++) {
+            (bytes32 orderId, bool isSettled, address inputToken, address settlementReceiver, uint256 amount) =
+                _decodeOrdersToSettle(requestIds[i], results[i]);
+
+            orderIds[i] = orderId;
+            areSettled[i] = isSettled;
+
+            if (isSettled) {
+                bool foundExistingUserTokenKey = false;
+                for (uint256 j = 0; j < uniqueUserTokenCount; j++) {
+                    if (userTokenKeys[j].receiver == settlementReceiver && userTokenKeys[j].token == inputToken) {
+                        amounts[j] += amount;
+                        foundExistingUserTokenKey = true;
+                        break;
+                    }
+                }
+
+                if (!foundExistingUserTokenKey) {
+                    userTokenKeys[uniqueUserTokenCount] = UserTokens({ receiver: settlementReceiver, token: inputToken });
+                    amounts[uniqueUserTokenCount] = amount;
+                    uniqueUserTokenCount++;
+                }
```

Comment:
this is super not readable for me and if I understand correctly it's just a gas optimization to aggregate multiple transfers to the same solver into one. Not work the smoke IMO. I wouldn't be surprised if it's actually more gas this way.